rpcserver.go

```diff
@@ -706,6 +709,30 @@ func (r *rpcServer) prepareSubServers(macService *macaroons.Service,
 		// interceptors below.
 		subServers = append(subServers, subServer)
 		subServerPerms = append(subServerPerms, macPerms)
+
+		if subServer.Name() == walletrpc.SubServerName {
```

Comment:
Hmmm I agree that this singling out of the `WalletKit` server isn't ideal and feels unclean. I think the cleanest i could without having to change the `InjectDependencies` interface just for this was to execute both `PopulateRemoteSignerCfgValues` & `InjectDependencies` for the `WalletKit` within this check, instead of doing it after.

So I updated the code to that. Let me know if you still don't like the current version, and have a better idea :)!